<launch>
  <arg name="set" default="set1"/>
  <arg name="robot" default="scitos"/>
  <arg name="num_of_plans" default="3000"/>
  <!-- <arg name="plan_length" default="25"/> -->
  <arg name="time_window" default="85"/>
  <arg name="rho" default="1.0"/>
  <arg name="input_file" default="$(find viper)/data/op/$(arg set)/views.json"/>
  <arg name="input_file_values" default="$(find viper)/data/op/$(arg set)/view_values.json"/>
  <arg name="input_file_costs" default="$(find viper)/data/op/$(arg set)/view_costs.json"/>


  <arg name="output_file" default="$(find viper)/data/op/$(arg set)/plans-$(arg rho)-$(arg time_window).json"/>


  <include file="$(find viper)/launch/$(arg robot)/_generate_plans.launch"/>

  <node name="generate_plans" pkg="viper" type="op_gen_plans.py" output="screen">
    <param name="robot" type="string" value="$(arg robot)"/>
    <param name="num_of_plans" type="int" value="$(arg num_of_plans)"/>
    <!-- <param name="plan_length" type="int" value="$(arg plan_length)"/>-->
    <param name="time_window" type="int" value="$(arg time_window)"/>
    <param name="rho"  type="double" value="$(arg rho)"/>
    <param name="input_file" type="string" value="$(arg input_file)"/>
    <param name="input_file_values" type="string" value="$(arg input_file_values)"/>
    <param name="input_file_costs"  type="string" value="$(arg input_file_costs)"/>
    <param name="output_file"       type="string" value="$(arg output_file)"/>
  </node>
</launch>
